### ISSUE TITLE
Proof of Concept for NetCDF Obs Data

### ISSUE THREAD

**Issue by kknee** [glos/myglos#5]
- [x] @lukecampbell provide Guan with sample netCDF file
- [x] Guan write netCDF file for a single GLOS Station to TDS using sample format
- [x] @lukecampbell review file for metadata compliance
- [ ]  @kkoch to get new endpoints (ncSOS & OPeNDAP) listed  in GN
- [x] @cheryldmorse to coordinate with Bob on adding  demo station to GLOS Portal for review

**Comment by cheryldmorse** [comment]
The contents of the glos_obs database is now being written to netCDF. The data is available [here](http://tds.glos.us/thredds/buoys/glos_buoys.html)

**Comment by kknee** [comment]
@kkoch can you try making a separte GN record for one station so that we can look at switching oceansmap over to using the netCDF data?

**Comment by kkoch** [comment]
Copy of 45169 buoy metadata.  Does not have a valid obscache link so you'll need to let me know if or what needs to be done to make that so it shows on dev. But does have categories, keywords, info link.

http://data.glos.us/metadata/srv/eng/main.home?uuid=1a478f2e-eafa-4925-9b67-3120a3276aef

**Comment by kknee** [comment]
doesn't need a valid obs cache link, but it does need a link to the TDS record.  @lukecampbell what does oceansmap need?  a link to the ncml?

**Comment by kknee** [comment]
@kkoch example links to add to GN:

1. GLOS ncSOS Service: http://tds.glos.us/thredds/sos/buoy_agg/utlcp/utlcp.ncml
2. GLOS OPeNDAP Service: http://tds.glos.us/thredds/dodsC/buoy_agg/utlcp/utlcp.ncml.html - note on this one that appending '.html' at the end makes this link readable in a browser, but folks actually wanting to hit the data via opendap programatically would want to truncate after '.ncml'.  Not sure which is the best to use...

**Comment by kknee** [comment]
checked the IOOS catalog (catalog.ioos.us) and they provide both the straight .ncml and the .ncml.html, differentiating them like this:

![image](https://user-images.githubusercontent.com/196868/27197857-e48ff4d8-51dd-11e7-8fb6-c7077e27ae31.png)

**Comment by kkoch** [comment]
@cheryldmorse @Bobfrat 

To test, I have added three links to the record for UTLCP in GN.  Once RPS verifies that this works to show the data in the portal, I can take the next step and add the corresponding links to all the buoy records.  This is in addition to the existing GLOS Data Cache (GeoJSON) link we are using now.

URL > http://tds.glos.us/thredds/sos/buoy_agg/utlcp/utlcp.ncml
Protocol > OGC-SOS Sensor Observation Service (OGC:SOS)
Name > GLOS ncSOS Service

URL > http://tds.glos.us/thredds/dodsC/buoy_agg/utlcp/utlcp.ncml
Protocol > OpeNDAP URL
Name > GLOS OPeNDAP Service

URL > http://tds.glos.us/thredds/dodsC/buoy_agg/utlcp/utlcp.ncml.html
Protocol > Web address (URL)
Name > GLOS OPeNDAP html interface

**Comment by Bobfrat** [comment]
@kkoch The values for lat and lon variables in this file seem to be swapped which is putting this station in Antarctica.

```
Dataset {
    Float64 lat;
    Float64 lon;
} buoy_agg/utlcp/utlcp.ncml;
---------------------------------------------
lat, -83.370224
lon, 41.724365
```

**Comment by kkoch** [comment]
Cool. Go figure I'd randomly pick a station with a problem. But hmmmm I have it right in my GN record and it is showing in the portal correctly from obs.  @gcutrell  do you know how to adjust in the .nc file to correct and how that data is gotten to us in the first place?

**Comment by cheryldmorse** [comment]
@kkoch That's an issue with the script that creates the nc files. I'll get it updated

**Comment by gcutrell** [comment]
@kkoch Looks like Cheryl sees the issue. It is fine in the obs database like you mentioned.

**Comment by cheryldmorse** [comment]
@Bobfrat The lat / lon variables should be correct now.

**Comment by Bobfrat** [comment]
Ok on http://dev.oceansmap.com/myglos/# you'll see this station listed as Demo Stations - UTLCP

**Comment by kkoch** [comment]
So what are our next steps?  It obviously works.  Seems like we would want to have some sort of full titles for the variables (e.g. Wind Direction vs. WDIR).  

@tslawecki > chime in?

![image](https://user-images.githubusercontent.com/6392982/30291729-bd5273e6-9701-11e7-92ab-75402031db16.png)

**Comment by Bobfrat** [comment]
Yes, so we use the netCDF variable names for display in the popup.

**Comment by cheryldmorse** [comment]
@Bobfrat I updated the variable names in the netCDF files. Can you regenerate the catalog?

**Comment by Bobfrat** [comment]
all set
<img width="530" alt="screen shot 2017-09-26 at 8 57 59 am" src="https://user-images.githubusercontent.com/5702672/30861509-ea490c50-a298-11e7-952e-aae2bd1f6c53.png">

**Comment by cheryldmorse** [comment]
@kkoch Can you update the GN record for utlcp to point to this: http://tds.glos.us/thredds/dodsC/buoy_agg_standard/utlcp/utlcp.ncml

**Comment by kkoch** [comment]
done

**Comment by cheryldmorse** [comment]
sorry @kkoch  I meant the 'Demo' utlcp station. Right now we need to keep the original netCDF files with the old variable names because the glbuoy portal is dependent on the variable names. I have the new netCDF files in a different catalog entry on the TDS

**Comment by kkoch** [comment]
I don't have a gn record for demo.

**Comment by kknee** [comment]
@cheryldmorse can you clarify what is need in GN here?  The GN GLOS OPeNDAP Service link points to the demo utlcp station on TDS, but the ncSOS does not

**Comment by cheryldmorse** [comment]
@Bobfrat - What GN record link are you using for the 'utlcp' station?

**Comment by kknee** [comment]
thanks to @Bobfrat  this is on dev now. 

_There's no valid data streaming but we're now fetching via opendap from the link provided in GN. If you move the map back to October you'll see data.

I'm grabbing the parameters from the GN record (descriptive keywords - variables) so those must match the variable names in the file. Thats why some of the variables have "no results". Or I can code it up to go fetch the dataset and parse the variable names._

**Comment by kkoch** [comment]
(Of course we would pick a station that isn't reporting so we could see real-ish-time.)   But we need a path forward/plan > I need to know specifically where to go and what to change so we don't break production.

The reason I have questions is if I look at the .nc file listing on tds for utlcp the naming doesn't always match what I see in Obs but some work and some don't (e.g., "Air Pressure" doesn't and the tds name seems to match in GN and tds.  Then conversely "dissolved_oxygen" does but looks like on tds it is "Dissolved Oxygen in Sea Water measured in mg/L" not the shorter term.  Thus I am probably not looking in the right place and need some guidance.
![image](https://user-images.githubusercontent.com/6392982/40367968-a23f831e-5da8-11e8-8daf-6d2dd815c98e.png)

Also > I think we need all the values to always show "no results" if there are no current values.  Note on the screenshot where we see some units like "NaN percent" instead:
![image](https://user-images.githubusercontent.com/6392982/40367796-3d5420cc-5da8-11e8-8cb8-ccb441b65d16.png)

**Comment by kkoch** [comment]
Note: I think long-term if we were not relying on GN for the variable names that would be better.  In other words if I think I understand @kknee what your above comment means, I think if OM used GN for any links, abstract, title, categories like it does now - basically all the things that make sense to be curated by me, BUT pulled the variable names from the real-time data that was being submitted, it would remove a manual and sometimes annoying process (e.g., when the PI one week decides the parameter is 'wind direction' and then the next week it is 'wind from direction').

**Comment by kkoch** [comment]
And adding more justification > we just had a PI comment that he'd added a parameter (sig. wave height) and glbuoys picked it up and our legacy portal picked it up, but because it was manual for me, it hadn't gotten added until this morning and is waiting on the portal refresh.  So we definitely need to get the portal to be automatic as well.

**Comment by Bobfrat** [comment]
@kkoch I will parse the variable names from the file. We do some "prettifying" on the front end so the parameters shown on OM are not an exact match to the variable names in the netCDF file.
